spec/models/manageiq/providers/nuage/vsd_client/vcr_specs/vsd_client_spec.rb

```diff
@@ -0,0 +1,990 @@
+      expect(policy_group).to be_nil
+    end
+
+    it "should return empty response for policy_groups for domain" do
+      policy_groups = nil
+      VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/policy_groups_for_domain_empty', :record => :new_episodes) do
+        policy_groups = @vsd_client.get_policy_groups_for_domain(DOMAIN_ID)
+      end
+      expect(policy_groups).to be_nil
+    end
+  end
+
+  it "should fail on wrong password" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_pass', :record => :new_episodes) do
+      expect do
+        described_class.new("https://#{@hostname}:8443/nuage/api/v5_0", @userid, 'wrong_password')
+      end.to raise_error(MiqException::MiqInvalidCredentialsError)
+    end
+  end
+
+  it "should fail on wrong username" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_user', :record => :new_episodes) do
+      expect do
+        described_class.new("https://#{@hostname}:8443/nuage/api/v5_0", 'wrong_user', @password)
+      end.to raise_error(MiqException::MiqInvalidCredentialsError)
+    end
+  end
+
+  it "should fail on wrong hostname" do
+    VCR.use_cassette(described_class.parent.name.underscore + '/vsd_client/wrong_hostname', :record => :new_episodes) do
+      expect do
+        described_class.new("https://wronghost:8443/nuage/api/v5_0", @userid, @password)
+      end.to raise_error(SocketError)
+    end
+  end
+
+  def assert_object_not_empty(object)
+    expect(object.length).to be > 0
+    expect(object.length).to be_truthy
+  end
+
+  def assert_enterprises(enterprises)
+    # test count
+    expect(enterprises.count).to be > 0
+
+    # test first enterprise object
+    assert_enterprise(enterprises.first)
+  end
+
+  def assert_enterprise(enterprise)
+    expect(enterprise).to include(
+      "children",
+      "parentType",
+      "entityScope",
+      "lastUpdatedBy",
+      "lastUpdatedDate",
+      "creationDate",
+      "name",
+      "description",
+      "avatarType",
+      "avatarData",
+      "floatingIPsQuota",
+      "floatingIPsUsed",
+      "allowTrustedForwardingClass",
+      "allowAdvancedQOSConfiguration",
+      "allowedForwardingClasses",
+      "allowGatewayManagement",
+      "enableApplicationPerformanceManagement",
+      "encryptionManagementMode",
+      "localAS",
+      "dictionaryVersion",
+      "allowedForwardingMode",
+      "owner",
+      "ID",
+      "parentID",
+      "externalID",
+      "customerID",
+      "DHCPLeaseInterval",
+      "enterpriseProfileID",
+      "receiveMultiCastListID",
+      "sendMultiCastListID",
+      "associatedGroupKeyEncryptionProfileID",
+      "associatedEnterpriseSecurityID",
+      "associatedKeyServerMonitorID",
+      "LDAPEnabled",
+      "LDAPAuthorizationEnabled",
+      "BGPEnabled",
+    )
+
+    # test for date values
+    assert_datetime(enterprise['lastUpdatedDate'])
+    assert_datetime(enterprise['creationDate'])
```

Comment:
I see we have many functions below to check parameter by parameter. May I suggest we group them into a "bulk" function:

```ruby
assert_datetimes(enterprise, %w(lastUpdatedDate creationDate))
assert_identifier_strings(enterprise, %w(ID lastUpdatedBy owner ...))
```

Having such "bulk" funcions implemented, number of lines would decrease notably.